lib/logstash/inputs/gelf.rb

```diff
@@ -90,12 +112,83 @@ def stop
   rescue IOError # the plugin is currently shutting down, so its safe to ignore theses errors
   end
 
+  private
+  def tcp_listener(output_queue)
+
+    @logger.warn("Starting gelf listener (tcp) ...", :address => "#{@host}:#{@port_tcp}")
+
+    if @tcp.nil?
+      @tcp = TCPServer.new(@host, @port_tcp)
+    end
+
+    while !@shutdown_requested
+      Thread.new(@tcp.accept) do |client|
+        @logger.debug? && @logger.debug("Gelf (tcp): Accepting connection from:  #{client.peeraddr[2]}:#{client.peeraddr[1]}")
+
+        begin
+          while !client.nil? && !client.eof?
+          
+            begin # Read from socket
+              @data_in = client.gets("\u0000")
+            rescue => ex
+              @logger.warn("Gelf (tcp): failed gets from client socket:", :exception => ex, :backtrace => ex.backtrace)
+            end 
+ 
+             if @data_in.nil?
+              @logger.warn("Gelf (tcp): socket read succeeded, but data is nil.  Skipping.")
+              next
+            end
+           
+            # data received.  Remove trailing \0
+            @data_in[-1] == "\u0000" && @data_in = @data_in[0...-1]
+            begin # Parse JSON
+              @jsonObj = JSON.parse(@data_in)
```

Comment:
there's no need to make this an instance variable, it can be local instead: `jsonObj`

Reply:
Done